File: project/4.EX_総合理解度チェック/java/java_ex4_ex/src/java_ex4_ex/Ex4_ex.java
package java_ex4_ex;

import java.util.Scanner;

public class Ex4_ex {
	// 実行時の引数 "5 nickels 4 quarters 2 dimes"
	public static void main(String[] args) {
		//イテレータを使ったループ
		double d = 0.0, sum = 0.0;
		for(String s : args) {
			
			switch( s.charAt(0) ) {
			case 'n':	// nickels
				sum += ( d * 5 );
				System.out.println( "5￠コインは" + (int)d + "枚" );
				break;
			case 'q':	// quarters
				sum += ( d * 25 );
				System.out.println( "25￠コインは" + (int)d + "枚" );
				break;
			case 'd':	// dimes
				sum += ( d * 10 );
				System.out.println( "10￠コインは" + (int)d + "枚" );
				break;
			default:	// 数値(硬貨の枚数)
				d = Double.valueOf( s );
			}
		}
		
		System.out.println( "コインの総額は＄" + sum/100 );
		System.out.println();
		
		// コンソール入力情報を判別
		System.out.print( "One/Two/Three/Four/Fiveのいずれかを入力：" );
		Scanner sc = new Scanner( System.in );
		String s = sc.nextLine();
		
		StringBuffer sb = new StringBuffer( " はスペイン語で " );
		sb.insert(0, s);
		if( s.equals( "One" ) ) {
			sb.append("Uno");
		}
		else if( s.equals( "Two" ) ) {
			sb.append("Dos");
		}
		else if( s.equals( "Three" ) ) {
			sb.append("Tres");
		}
		else if( s.equals( "Four" ) ) {
			sb.append("Quatro");
		}
		else {
			sb.append("Cinco");
		}
		
		System.out.println( sb );
		
		
		sc.close();
	}

}

File: project/4.EX_総合理解度チェック/py/ex4_ex/src/ex4_ex.py
'''
Created on 2023/03/29

@author: Keroichi.T
'''

import sys

# 実行時の引数 "5 nickels 4 quarters 2 dimes"
def main():
    d, s = 0, 0.
    for arg in sys.argv[1:]:
        match arg[0]:
            case "n":
                s += (d * 5)
                print("5￠コインは {} 枚".format(d))
            case "q":
                s += (d * 25)
                print("25￠コインは {} 枚".format(d))
            case "d":
                s += (d * 10)
                print("10￠コインは {} 枚".format(d))
            case _:         # ワイルドカード→数値(硬貨の枚数)
                d = int(arg)
    
    print("コインの総額は＄{}".format(s/100), end="\n\n")
    
    # コンソール入力情報を判別する
    # ↓ディクショナリ型(組込み型)
    dic = {"One":"Uno",         
           "Two":"Dos",
           "Three":"Tres",
           "Four":"Quatro",
           "Five":"Cinco"}
    cin = input("One/Two/Three/Four/Fiveのいずれかを入力：\n >> ")

    msg = [""," はスペイン語で ",""]
    msg[0], msg[2] = cin, dic[cin]
    print("".join(msg))

if __name__ == '__main__':
    main()